components/prism-factor.js

```diff
@@ -0,0 +1,277 @@
+			lookbehind: true,
+			alias: 'function',
+			inside: {
+				// this is essentially a regex for vocab names, which i don't want to specify
+				// but the USING: gets picked up as a vocab name
+				string: {
+					pattern: /(\s)[^:\r\n\t ]+/,
+					lookbehind: true
+				}
+			}
+		},
+
+		/* this description of stack effect literal syntax is not complete and not as specific as theoretically possible
+		 	trying to do better is more work and regex-computation-time than it's worth though.
+			- we'd like to have the "delimiter" parts of the stack effect [ (, --, and ) ] be a different (less-important or comment-like) colour to the stack effect contents
+			- we'd like if nested stack effects were treated as such rather than just appearing flat
+			- we'd like if the following variable name conventions were recognised specifically:
+				special row variables = ..a b..
+				nested quotation variables end with a colon = quot: ( a: ( -- ) -- b )
+				word throws unconditional error = *
+				any other word-like variable name = a ? q' etc
+
+			https://docs.factorcode.org/content/article-effects.html
+
+			these are pretty complicated to highlight properly without a real parser, and therefore out of scope
+		'stack-effect-old': {
+			pattern: /(^|\s)(?:call|execute|eval)?\((?:\s+[^"\r\n\t ]\S*)*?\s+--(?:\s+[^"\n\t ]\S*)*?\s+\)(?=\s|$)/,
+			lookbehind: true,
+			alias: 'operator',
+		},
+
+		TODO:
+		'stack-effect-variable': {
+			{ pattern: / /, lookbehind: true, alias: 'operator' },
+		},
+		*/
+
+		// current solution is not great
+		'stack-effect-open': {
+			pattern: /(^|\s)(?:call|execute|eval)?\((?=\s)/,
+			lookbehind: true,
+			alias: 'operator' },
+		'stack-effect-middle': {
+			pattern: /(\s)--(?=\s)/,
+			lookbehind: true,
+			alias: 'operator' },
+		'stack-effect-close': {
+			pattern: /(\s)\)(?=\s|$)/,
+			lookbehind: true,
+			alias: 'operator' },
+
+		combinators: [],
+
+		'kernel-builtin': {
+			pattern: null,
+			lookbehind: true,
+			alias: 'variable'
+		},
+
+		'sequences-builtin': {
+			pattern: null,
+			lookbehind: true,
+			alias: 'variable'
+		},
+
+		'math-builtin': {
+			pattern: null,
+			lookbehind: true,
+			alias: 'variable'
+		},
+
+		'constructor-word': {
+			// <array> but not <=>
+			pattern: /(^|\s)<(?!=+>|-+>)[^\r\n\t ]+>(?=\s|$)/,
+			lookbehind: true,
+			alias: 'keyword'
+		},
+
+		'other-builtin-syntax': {
+			pattern: null,
+			lookbehind: true,
+			alias: 'operator'
+		},
+
+		'conventionally-named-word': {
+			// for certain word naming conventions like +lt+, sequence?, readers>>, >>setters, writers<<
+			pattern: /(^|\s)(?:set-[^\r\n\t ]+|change-[^\r\n\t ]+|>[^>\r\n\t ]+|[^:>\r\n\t ]+>|[^>\r\n\t ]+>[^>\r\n\t ]+|\+[^\+\r\n\t ]+\+|[^?\r\n\t ]+\?|[^>\r\n\t ]+>>|>>[^>\r\n\t ]+|[^<\r\n\t ]+<<|\?[^?\r\n\t ]+|\([^()\r\n\t ]+\)|[^!\r\n\t ]+!)(?=\s|$)/,
+			lookbehind: true,
+			alias: 'keyword'
+		},
+
+		'colon-syntax': {
+			pattern: /(^|\s)(?:[A-Z]*#?):{1,2}\s+(?:(?!;)|;\S+|[^\r\n\t; ]\S*)(?=\s|$)/,
```

Comment:
The goal of this is to disallow `: ;` (because semicolon `;` is a definition-ender, not a word name, and shouldn't be highlighted as valid), but to allow `: a;` and `: ;a` because  `a;` and `;a` are word names; `;` is only a definition-ender when it is alone as a token, not part of another token.

You are right, I didn't read what I had written and the correct regex is as you say.

Reply:
It's ok. Especially when rewriting patterns, these things can happen.